### DISCUSSION TITLE
Use of reserved identifiers

### DISCUSSION THREAD

**Discussion by errorhandler** [google/cel-go#1008] [Q&A]
Hi there,

I'm trying to implement a custom CEL function that provides alternative syntax for conditionals, via a excel style `if` function. However diving into the cel implementation, `if` is a reserved identifier. I have a couple of questions:

1. What's the reason for the set of reserved identifiers?
2. Is there anyway around this?'

I've implemented it using different casing and it works well however it would be nice to be able to match the casing for other CEL functions.

Thanks in advance!

**Comment by TristonianJones** [comment]
The concern is mostly if CEL chooses at some point to support such keywords, or if another language extending CEL does. 

Instead of using `if` consider may using `when`?

-Tristan
